test/ui_test_suite.rb

```diff
@@ -2732,6 +2971,454 @@ def compare_within_rounding_error(error, a1, a2)
+		name.clear
+		name.send_key("user-#{$random_seed}")
+
+		#revert labels
+		annotation_labels = @driver.find_elements(:id, 'user-annotation_values')
+
+		annotation_labels.each_with_index do |annot, i|
+			annot.clear
+			annot.send_keys("group#{i}")
+		end
+
+		#update annotation
+		submit = @driver.find_element(:id, 'submit-button')
+		submit.click
+
+		wait_until_page_loads('user annotation path')
+
+		#check new names and labels
+		new_names = @driver.find_elements(:class, 'annotation-name').map{|x| x.text }
+		new_labels = @driver.find_elements(:class, "user-#{$random_seed}").map{|x| x.text }
+
+		#assert new name saved correctly
+		assert !(new_names.include? "user-#{$random_seed}new"), "Name edit failed, expected 'user-#{$random_seed}' but got '#{new_names}'"
+
+		#assert labels saved correctly
+		assert !(new_labels.include? "group0new"), "Name edit failed, did not expect 'new in group' but got '#{new_labels}'"
+
+		wait_for_render(:id, 'message_modal')
+		close_modal('message_modal')
+
+		#View the annotation
+		@driver.find_element(:class, "user-#{$random_seed}-show").click
+		wait_until_page_loads('view user annotation path')
+
+		#assert the plot still renders
+		@wait.until {wait_for_plotly_render('#cluster-plot', 'rendered')}
+		annot_rendered = @driver.execute_script("return $('#cluster-plot').data('rendered')")
+		assert annot_rendered, "cluster plot did not finish rendering on annotation change, expected true but found #{annot_rendered}"
+
+		#assert labels are correct
+		plot_labels = @driver.find_elements(:class, "user-select-none").map{|x| x.attribute('data-unformatted') }
+		assert (plot_labels.include? "user-#{$random_seed}: group0 (3 points)"), "labels are incorrect: '#{plot_labels}' should include 'user-#{$random_seed}: group0'"
+	end
+
+	test 'front-end: download annotation cluster file' do
+		puts "Test method: #{self.method_name}"
+		login_path = @base_url + '/users/sign_in'
+		# downloads require login now
+		@driver.get login_path
+		wait_until_page_loads(login_path)
+		login($test_email)
+
+		# load annotation panel
+		annot_path = @base_url + '/user_annotations'
+		@driver.get annot_path
+
+		#Click download and get filenames
+		filename = (@driver.find_element(:class, "user-#{$random_seed}_cluster").attribute('innerHTML') + "_user-#{$random_seed}.txt").gsub(/ /, '_')
+		basename = filename.split('.').first
+
+		@driver.find_element(:class, "user-#{$random_seed}-download").click
+		# give browser 5 seconds to initiate download
+		sleep(5)
+		# make sure file was actually downloaded
+		file_exists = Dir.entries($download_dir).select {|f| f =~ /#{basename}/}.size >= 1 || File.exists?(File.join($download_dir, filename))
+		assert file_exists, "did not find downloaded file: #{filename} in #{Dir.entries($download_dir).join(', ')}"
+
+		# delete file
+		File.delete(File.join($download_dir, filename))
+
+		puts "Test method: #{self.method_name} successful!"
+	end
+
+	#check user annotation deletion
```

Comment:
This test has no assertions, so therefore it is invalid.  You should grab a DOM id for an element, delete it, and then confirm it has been removed.  You should also check that it's been removed from the list of available annotations on the view_study page.